list top artists/playlist, add mime types, fallback to default mime type

mgoltzsche opened the pull request:
* Aggregates the top artists per playlist and exposes them comma-separated within the `comment` field of each playlist. Also optimizes direct/single playlist access.
* Map mime types for m4a, mp4 and opus file extensions.
* When no mime type registered for file extension, fallback to serving the file using mime type `application/octet-stream` (and log a warning) instead of failing the request. (At least GStreamer/Mopidy seems to do its own format detection anyway since it still plays such streams.)

mgoltzsche committed:
feat(playlists): aggregate top artists

Aggregates the top artists per playlist and exposes them comma-separated within the `comment` field of each playlist.
Also optimizes direct/single playlist access.

mgoltzsche committed:
fix: add mime types, fallback to default mime type

* Map .m4a, .mp4 and .opus file extensions to mime types.
* Fallback to using mime type 'application/octet-stream' instead of failing the request when no mime type registered for file extension.

mgoltzsche committed:
fix: group artists by name without accents

Using the lower-case artist name without accents as key when grouping artists within a playlist.

mgoltzsche commented:
@BinaryBrain it would be great if you could review and merge this PR soon. Not having the mime type fix/enhancement prevents me from serving my music library using upstream Beetstream and I guess others have the same problem, e.g. see #11.

BinaryBrain commented:
Yes, I've been quite busy recently and the current version is broken if you don't have the smartplaylist plugin.

mgoltzsche commented:
ah, right, it fails [here](https://github.com/BinaryBrain/Beetstream/blob/20b0a9e5a7319b221bf7c082522c1d7221cfe48f/beetsplug/beetstream/__init__.py#L83) when neither the smartplaylist plugin was enabled nor the `playlist_dir` option specified. I'll add a commit to this PR to fix it...

mgoltzsche committed:
fix: don't fail on missing playlist configuration

Don't fail but log a warning when the smartplaylist beets plugin is not enabled and the `playlist_dir` configuration option was not specified.

mgoltzsche commented:
Done: Now I've pushed another commit to not fail but log a warning when the smartplaylist plugin is not enabled and the `playlist_dir` configuration option was not specified.
